compiler/rustc_typeck/src/coherence/mod.rs

```diff
@@ -75,6 +75,19 @@ fn enforce_trait_manually_implementable(
         return;
     }
 
+    if did == li.as_repr_trait() {
+        let span = impl_header_span(tcx, impl_def_id);
+        struct_span_err!(
+            tcx.sess,
+            span,
+            E0322,
```

Comment:
The message for this code explicitly mentions `Sized`. Use a new error code and add a test.